stack_overflow/src/main/java/com/gabrielcireap/stackOverflow/repository/jdbc/JdbcQuestionRepository.java

```diff
@@ -75,22 +75,17 @@ public void remove(Question question) {
     }
 
     @Override
-    public List<Question> findByTag(String tag) {
+    public List<Question> findByTag(Tag tag) {
         List<Question> questions =  template.query(
         "SELECT question.id, question.user_id, question.title, question.text, question.creation_date, question.vote_count, tag.id as \"tag_id\", tag.name as \"tag_name\", " +
             "user.id, user.username, user.password, user.email, user.score, user.is_admin, user.is_banned" +
             " FROM question JOIN questiontotags ON question.id = questiontotags.question_id " +
             "JOIN tag ON tag.id = questiontotags.tag_id " +
-            "JOIN user ON user.id = question.user_id WHERE tag.name = '" + tag + "'", new QuestionMapper());
+            "JOIN user ON user.id = question.user_id WHERE tag.name = '" + tag.getName() + "'", new QuestionMapper());
```

Comment:
This is a place in which SQL Injection can easily occur. Use prepared statement parameters (`?`) instead and pass the tag name to the template as a parameter (like you do it in the other queries).